
message SdkInvalidOperation {
  required int32 opcode = 1;
  required int32 what = 2;
  required bytes why = 3;
}

message SdkSet {
  required int32 opcode = 1;
  required bytes table = 2;
  required bytes key = 3;
  required bytes value = 4;
}

message SdkSetRet {
  required int32 opcode = 1;
  required bool status = 2;
}

message SdkCas {
  required int32 opcode = 1;
  required bytes table = 2;
  required bytes key = 3;
  required bytes value = 4;
  required int32 version = 5;
}

message SdkCasRet {                                                                                                                                                                                                                
  required int32 opcode = 1;
  required bool status = 2;
}

message SdkGet {
  required int32 opcode = 1;
  required bytes table = 2;
  required bytes key = 3;
}

message SdkGetRet {
  required int32 opcode = 1;
  required bytes value = 2;
}

message SdkGetV {
  required int32 opcode = 1;
  required bytes table = 2;
  required bytes key = 3;
}

message SdkGetVRet {
  required int32 opcode = 1;
  required bytes value = 2;
  required int32 version = 3;
}

message SdkDelete {
  required int32 opcode = 1;
  required bytes table = 2;
  required bytes key = 3;
}

message SdkDeleteRet {
  required int32 opcode = 1;
  required bool status = 2;
}

message SdkPing {
  required int32 opcode = 1;
}

message SdkPingRet {
  required int32 opcode = 1;
  required bool status = 2;
}

message SdkGetIfAll {
  required int32 opcode = 1;
  required bytes table = 2;
  required bytes key = 3;
  required bool Strict = 4;
}

message SdkGetIfAllRet {
  required int32 opcode = 1;
  required bytes value = 2;
  required int32 version = 3;
}

message SdkMGet {
  required int32 opcode = 1;
  required bytes table = 2;
  repeated bytes key = 3;
}

message SdkMGetRet {
  required int32 opcode = 1;

  message KeyValue {
    required bytes key = 1;
    required bytes value = 2;
    required int32 status = 3;
  }

  repeated KeyValue rets = 2; 
}






